spyder/plugins/onlinehelp/widgets.py

```diff
@@ -22,6 +23,189 @@
 from spyder.widgets.browser import WebBrowser
 
 
+# =============================================================================
+# Pydoc adjustments
+# =============================================================================
+# This is needed to prevent pydoc raise an ErrorDuringImport when
+# trying to import numpy.
+# See spyder-ide/spyder#10740
+DIRECT_PYDOC_IMPORT_MODULES = ['numpy', 'numpy.core']
+try:
+    from pydoc import safeimport
+
+    def spyder_safeimport(path, forceload=0, cache={}):
+        if path in DIRECT_PYDOC_IMPORT_MODULES:
+            forceload = 0
+        return safeimport(path, forceload=forceload, cache=cache)
+
+    pydoc.safeimport = spyder_safeimport
+except Exception:
+    pass
+
+
+# A reimplementation of pydoc._start_server to handle connection errors
+# for 'do_GET'.
+def _start_server(urlhandler, hostname, port):
+    """Start an HTTP server thread on a specific port.
+
+    Taken from PyDoc: https://github.com/python/cpython/blob/3.7/Lib/pydoc.py
+    Start an HTML/text server thread, so HTML or text documents can be
+    browsed dynamically and interactively with a Web browser.  Example use:

```

Comment:
Please remove all contents of this docstring below this line. I think they are not necessary because they from the original `_start_server`, right?

Reply:
Also, let's rewrite this docstring like this

```python
"""
Start an HTTP server thread on a specific port.

This is a reimplementation of `pydoc._start_server` to handle connection
errors for 'do_GET'.

Taken from PyDoc: https://github.com/python/cpython/blob/3.7/Lib/pydoc.py
"""
```